webapp/channels/src/components/team_groups_manage_modal/team_groups_manage_modal.test.tsx

```diff
@@ -0,0 +1,112 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {shallow} from 'enzyme';
+import React from 'react';
+
+import TeamGroupsManageModal from 'components/team_groups_manage_modal/team_groups_manage_modal';
+
+import {TestHelper} from 'utils/test_helper';
+import { shallowWithIntl } from 'tests/helpers/intl-test-helper';
+
+describe('components/TeamGroupsManageModal', () => {
+    const team = TestHelper.getTeamMock({ type: 'O', allowed_domains: '', allow_open_invite: false, scheme_id: undefined });
+    const actions = {
+        getGroupsAssociatedToTeam: jest.fn().mockResolvedValue({data: {groups: [], totalGroupCount: 0}}),
+        closeModal: jest.fn(),
+        openModal: jest.fn(),
+        unlinkGroupSyncable: jest.fn().mockResolvedValue({data: []}),
+        patchGroupSyncable: jest.fn().mockResolvedValue({data: []}),
+        getMyTeamMembers: jest.fn(),
+    };
+
+    const baseProps = {
+        intl: {
+            formatMessage: jest.fn(),
+        },
+        team,
+        actions
+    };
+
+    test('should match snapshot', () => {
+        const wrapper = shallow(<TeamGroupsManageModal {...baseProps}/>);
+        expect(wrapper).toMatchSnapshot();
+    });
+
+    test('should hide confirm modal when team changes', () => {
+        const wrapper = shallow(<TeamGroupsManageModal {...baseProps}/>);
+        wrapper.setProps({team: TestHelper.getTeamMock({id: 'new'})});
+        expect(wrapper).toMatchSnapshot();
+    });
+
+    test('should call getGroupsAssociatedToTeam on loadItems', async () => {
+        const wrapper = shallowWithIntl(<TeamGroupsManageModal {...baseProps}/>);
+        const instance = wrapper.instance() as any;
+        await instance.loadItems(0, '');
+        expect(actions.getGroupsAssociatedToTeam).toHaveBeenCalledTimes(1);
+    });
+
+    test('should call handleDeleteCanceled', () => {
+        const wrapper = shallowWithIntl(<TeamGroupsManageModal {...baseProps}/>);
+        const instance = wrapper.instance() as any;
+        instance.handleDeleteCanceled();
+        expect(wrapper.state('showConfirmModal')).toBe(false);
+    });
+
+    test('should call handleDeleteConfirmed', async () => {
+        const wrapper = shallowWithIntl(<TeamGroupsManageModal {...baseProps}/>);
+        const instance = wrapper.instance() as any;
+        await instance.handleDeleteConfirmed();
+        expect(actions.unlinkGroupSyncable).toHaveBeenCalledTimes(1);
+    });
+
+    test('should match snapshot when onClickRemoveGroup is called', () => {
+        const wrapper = shallowWithIntl(<TeamGroupsManageModal {...baseProps}/>);
+        const instance = wrapper.instance() as any;
+        instance.onClickRemoveGroup({id: 'group_id'}, {setState: jest.fn()});
+        expect(wrapper).toMatchSnapshot();
```

Comment:
same comment here than what @larkox mentioned previously, toMatchSnapshot is not the best approach for this situation. Here it makes more sense to check for elements in the rendered container.

Reply:
Sure @pvev I have accomodated all your points in my latest commit on this PR. Please check. Also I have a short comment on the implementation of ConfirmModal component which I have posted in the Mattermost Developers Channel as a reply to my PR merge request post. Kindly check it at: https://community.mattermost.com/core/pl/p5ydjbjkbtb88k1fhgq8uqhpuw